Enable download of datapackage as zip

Bachibouzouk opened the pull request:
This enable the download of a datapackage of a scenario. 

One can use the URL:
http://127.0.0.1:8000/en/scenario/export/datapackage/<id of the scenario to export>

Or click on "Export as datapackage" on the scenario step 2 footer

TODO: 

- [x] Generate the datapackage.json automatically as well
- [x] Get the foreign keys of datapackage.json correctly
- [x] Figure out a parameter matching between open_plan parameters and the placades parameters --> this is done in https://github.com/open-plan-tool/placades/pull/5

Bachibouzouk commented:
The idea is to enable download of datapackage and then iterate to make changes to the `to_datapackage` method

paulapreuss reviewed (commented):
Using the current pinned version of tabular in the requirements (pointing to the `dev` branch), I get the error `infer_metadata_from_data() got an unexpected keyword argument 'fk_targets'`. I looked around the `oemof-tabular` repo and saw your PR which introduces this keyword in `fix/datapackage`. However, pointing to this branch instead causes the following dependency conflict in the requirements:

```
ERROR: Cannot install datapackage==1.5.1 and jsonschema==4.4.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested jsonschema==4.4.0
    datapackage 1.5.1 depends on jsonschema<3.0 and >=2.5
```

I was able to bypass this error by unpinning the `jsonschema` version, but now I am not sure if the version installed might cause issues within the functions that use the package. Would have to double check this. 

After this, the datapackage export seems to work and contain all the necessary files. Very cool feature!

Bachibouzouk commented:
> I was able to bypass this error by unpinning the `jsonschema` version, but now I am not sure if the version installed might cause issues within the functions that use the package. Would have to double check this.

I think we should now soon be able to upgrade the datapackage package in oemof-tabular. Moreover we decided with @uvchik it would be more convenient to create https://github.com/oemof/oemof-datapackage instead of making breaking changes in oemof-tabular. I will focus now on making the datapackages downloaded from OpenPlan app run using placade

Bachibouzouk commented:
Note @paulapreuss : the structure of the Storage Assets is outdated, as soon as we simulate using placades we can simplify the storages and bring their structure one to one with the fields of the corresponding placades

Bachibouzouk committed:
Update oemof-thermal and add oemof-datapackage in requirements

Bachibouzouk committed:
REMOVE COMMIT Install git within Dockerfile

I think this was because I first used installs from github with pip,
this is likely not needed anymore

Bachibouzouk committed:
Add a view to export datapackage as zip files

Bachibouzouk committed:
Add project information to datapackage

Bachibouzouk committed:
Populate the elements folder of the datapackage

Bachibouzouk committed:
Add unique busses as a resource in the elements folder of the datapackage

Bachibouzouk committed:
Add all the profiles as a single resource in the sequences folder of the datapackage

Bachibouzouk committed:
Infer a datapackage.sjon file from the datapackage

Bachibouzouk committed:
Alter to_datapackage method for Bus class

Bachibouzouk committed:
Bind project data for assets which can be optimized

Bachibouzouk committed:
Export Storage asset to datapackage

Their structure with children asset is cumbersome
and should be adapted in the database

Bachibouzouk committed:
Export special assets to datapackage

CHP, heat pump and electrolyzer

Bachibouzouk committed:
Adapt assets' input and output port names

Bachibouzouk committed:
Adapt datapackagege export command line

python manage.py datapackage <scen number> -o <output path>

Bachibouzouk committed:
Add a button in step 2 to export scenario as datapackage

Bachibouzouk commented:
@paulapreuss - everything is ready to be reviewed. Currently there is still the error you mentionned and I plan to fix it by following the changes made in https://github.com/oemof/oemof-tabular/pull/188 in order to free ourselves from the datapackage dependency problem

Bachibouzouk commented:
@paulapreuss, @uvchik - I fixed the installation problem in a branch based on https://github.com/oemof/oemof-datapackage/pull/1. However I don't have the rights to push it to oemof-datapackage repo. Once this is merged, we should have a pypi installable version of oemof-datapackage (@paulapreuss this is why we currently still need [9d53876](https://github.com/open-plan-tool/gui/pull/422/commits/9d538767bfc0fe3fc98651c9bcc3525021fdda48)). Locally you need to install `pip install git+https://github.com/oemof/oemof-datapackage.git@fix/infer_datapackage` (once `fix/infer_datapackage` is pushed as I get the rights)

paulapreuss reviewed (approved):
This is really, really nice 😀 I tested around a bit and found that it works really nicely, I also had no issues with the depency install. Imo, this is ready to be merged to main and tested on the staging server.

Bachibouzouk commented:
Thanks @paulapreuss ! I am glad it went frictionless for you this time!

@uvchik - do you think we could have a release of oemof-datapackage already, or it is better to wait until the meeting with the other parties in the discussion before making a release? I am asking because of the last commit of this PR, it is not useful if there is a pypi release and I would remove it. If the release cannot be made quickly, then I will keep the commit to install oemof-datapackage from the git branch
